### PR TITLE
Fix csrmm2 to support transa

### LINKED ISSUE #552: csrmm2 crashes when using transa=True
When running the following code
```
import numpy as np
import scipy.sparse as sp

from chainer import cuda

import cupy
import cupy.sparse

def main():
    x = np.random.uniform(-1, 1, (4, 3)).astype(np.float32)
    x = cuda.to_gpu(x)
    W = sp.csr_matrix(np.random.uniform(-1, 1, (2, 3))).astype(np.float32)
    W = cupy.sparse.csr_matrix(W)

    y = cupy.cusparse.csrmm2(W, x.T).T

    gx = cupy.cusparse.csrmm2(W, y.T, transa=True).T  # This does not work

if __name__ == '__main__':
    main()
```
cupy crashes with the error:
```
Traceback (most recent call last):
  File "main.py", line 23, in <module>
    main()
  File "main.py", line 20, in main
    gx = cupy.cusparse.csrmm2(W, y.T, transa=True).T
  File "/home/tommi/.pyenv/versions/cupy-dev/lib/python3.4/site-packages/cupy-2.0.0rc1-py3.4-linux-x86_64.egg/cupy/cusparse.py", line 205, in csrmm2
    b.data.ptr, ldb, beta.data, c.data.ptr, ldc)
  File "/home/tommi/.pyenv/versions/cupy-dev/lib/python3.4/site-packages/cupy-2.0.0rc1-py3.4-linux-x86_64.egg/cupy/cusparse.py", line 52, in _call_cusparse
    return f(*args)
  File "cupy/cuda/cusparse.pyx", line 347, in cupy.cuda.cusparse.scsrmm2
  File "cupy/cuda/cusparse.pyx", line 357, in cupy.cuda.cusparse.scsrmm2
  File "cupy/cuda/cusparse.pyx", line 229, in cupy.cuda.cusparse.check_status
cupy.cuda.cusparse.CuSparseError: CUSPARSE_STATUS_INVALID_VALUE
```
This can be fixed by replacing the crashing line with `gx = cupy.cusparse.csrmm2(W.T, y.T).T`, so there appears to be a bug.

I am using cupy 2.0.0rc1 and CUDA 8.0.

### PR DESCRIPTION
**Pull request by unnonouno**
I noticed that I misunderstood arguments `m` and `n` in `csrmm2` especially when `transa=True`.
fix #552

### CODE REVIEW

File: tests/cupy_tests/test_cusparse.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        else:
+            self.a = self.op_a
+        self.b = numpy.random.uniform(-1, 1, (3, 4)).astype(self.dtype)
+        self.c = numpy.random.uniform(-1, 1, (2, 4)).astype(self.dtype)
+
+    def test_csrmm(self):
+        a = cupy.sparse.csr_matrix(self.a)
+        b = cupy.array(self.b, order='f')
+        y = cupy.cusparse.csrmm(a, b, alpha=self.alpha, transa=self.transa)
+        expect = self.alpha * (self.op_a.dot(self.b))
```
Comment (takagi) [inline tests/cupy_tests/test_cusparse.py]: Please remove the outer parentheses if no special reason.

File: tests/cupy_tests/test_cusparse.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        testing.assert_array_almost_equal(y, expect)
+
+    def test_csrmm_with_c(self):
+        a = cupy.sparse.csr_matrix(self.a)
+        b = cupy.array(self.b, order='f')
+        c = cupy.array(self.c, order='f')
+        y = cupy.cusparse.csrmm(
+            a, b, c=c, alpha=self.alpha, beta=self.beta, transa=self.transa)
+        expect = self.alpha * (self.op_a.dot(self.b)) + self.beta * self.c
```
Comment (takagi) [inline tests/cupy_tests/test_cusparse.py]: Please remove the outer parentheses if no special reason.

File: tests/cupy_tests/test_cusparse.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        else:
+            self.b = self.op_b
+        self.c = numpy.random.uniform(-1, 1, (2, 4)).astype(self.dtype)
+
+    def test_csrmm2(self):
+        a = cupy.sparse.csr_matrix(self.a)
+        b = cupy.array(self.b, order='f')
+        y = cupy.cusparse.csrmm2(
+            a, b, alpha=self.alpha, transa=self.transa, transb=self.transb)
+        expect = self.alpha * (self.op_a.dot(self.op_b))
```
Comment (takagi) [inline tests/cupy_tests/test_cusparse.py]: Please remove the outer parentheses if no special reason.

File: tests/cupy_tests/test_cusparse.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        testing.assert_array_almost_equal(y, expect)
+
+    def test_csrmm2_with_c(self):
+        a = cupy.sparse.csr_matrix(self.a)
+        b = cupy.array(self.b, order='f')
+        c = cupy.array(self.c, order='f')
+        y = cupy.cusparse.csrmm2(
+            a, b, c=c, alpha=self.alpha, beta=self.beta,
+            transa=self.transa, transb=self.transb)
+        expect = self.alpha * (self.op_a.dot(self.op_b)) + self.beta * self.c
```
Comment (takagi) [inline tests/cupy_tests/test_cusparse.py]: Please remove the outer parentheses if no special reason.

**Comment by takagi** [comment]
LGTM!

**Comment by unnonouno** [comment]
fixed

**Comment by takagi** [comment]
jenkins test this please.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by takagi** [comment]
LGTM!

**Pull request merged**